plugins/gtav/structs.h

```diff
@@ -0,0 +1,227 @@
+// Copyright 2022 The Mumble Developers. All rights reserved.
+// Use of this source code is governed by a BSD-style license
+// that can be found in the LICENSE file at the root of the
+// Mumble source tree or at <https://www.mumble.info/LICENSE>.
+
+#ifndef GTAV_STRUCTS
+#define GTAV_STRUCTS
+
+#include <cstdint>
+
+// Reference: https://github.com/Yimura/GTAV-Classes
+
+typedef uint64_t ptr_t;
+
+enum class GameState : int32_t {
+	Invalid = -1,
+	Playing,
+	Died,
+	Arrested,
+	FailedMission,
+	LeftGame,
+	Respawn,
+	InMPCutscene
+};
+
+enum class ModelType : uint8_t { Unk0, Object, Unk1, Unk2, Weapon, Vehicle, Ped, Plant = 129 };
+
+namespace rage {
+template< typename T > union vector3 {
+	T data[3];
+	struct {
+		T x, y, z;
+	};
+};
+
+typedef vector3< float > fvector3;
+
+struct netAddress {
+	uint8_t field4;
+	uint8_t field3;
+	uint8_t field2;
+	uint8_t field1;
+};
+
+struct netPlayerData {
+	uint8_t pad1[8];
+	uint64_t rockstarID;
+	uint8_t pad2[52];
+	netAddress relayIP;
+	uint16_t relayPort;
+	uint8_t pad3[2];
+	netAddress externalIP;
+	uint16_t externalPort;
+	uint8_t pad4[2];
+	netAddress internalIP;
+	uint16_t internalPort;
+	uint8_t pad5[6];
+	uint64_t hostToken;
+	uint64_t peerID;
+	uint64_t rockstarID2;
+	uint8_t pad6[12];
+	char name[20];
+};
+
+#pragma pack(push, 4)
```

Comment:
Do we really require non-standard pragmas here? I would prefer to not use those in our code base

Reply:
Unfortunately required, otherwise the struct's size is wrong.

Reply:
Alright then - are these pragmas supported by all relevant compilers?

Reply:
Well, CI succeeding implies that Clang, GCC and MSVC support them.

Reply:
Afaik unknown pragmas are usually just silently ignored. I take it though that in case the pragma was ignored, the corresponding static_assert further down would be triggered, right?

Reply:
Technically it only means that the struct size is correct, but I would expect all compilers to align members the same way.

Reply:
Yeah, I think this should be a reasonably safe assumption